### ISSUE TITLE
Ctrl + Backspace doesn't work

### ISSUE THREAD

**Issue by MrDemocracy** [mridgers/clink#501]
Normally when I hold `Ctrl` and use the backspace key, it deletes everything until the next space, but with clink this doesn't happen.
The `Ctrl` + `←`/`→` (arrow keys) function does however works almost the same way as the stock cmd, but Clink also jumps to the next special character or any type of opening bracket. I wish there was an option to disable/enable this, so that it only jumps to the next space.

**Comment by MrDemocracy** [comment]
<kbd>CTRL</kbd>+<kbd>⌫</kbd> only deletes one character. It doesn't matter what text is put in the cmd prompt.

**Example with any type of brackets:**
`somerandomtext{somerandomtextinsquirlybrackets} .`
When I hit <kbd>CTRL</kbd>+<kbd>←</kbd> when the cursor is at the far right before the period, the cursor is moved here:
`somerandomtext{`(here)`somerandomtextinsquirlybrackets} .`
It's the same with () and [].
Why is clink even jumping to special characters? It's way better to jump to previous whitespaces, at least when using <kbd>CTRL</kbd>+<kbd>⌫</kbd>.

**Comment by aniketmeshram** [comment]
You are right. my bad, I'm looking into it. thanks

**Comment by gohar94** [comment]
Having the same issue with `CTRL` + backspace. Clink offers `ALT` + backspace instead which is not really what I am used to. :(

**Comment by altimmons** [comment]
I disagree with Mr. Democracy.  The appropriate option is to stop at the special character.  In fact I would be surprised if **base** command prompt didn't do that.  It's because in nearly every case I can think of, it denotes a new "token" of some sort.  Like `hamburger(8)`.  The appropriate behavior is to stop before the paren so that you can change the numerical term, and then hitting it again, change the word term.  This is extremely common.  So much so, that I don't think it is CLINK you are complaining of.  I think its windows.  If you are using braces like this `hamb}urg{er8(`  - then I hate to tell you, but you are doing it wrong.  **It should stop at each such delim**.  Think about code `public static void main(String[] args) {`  The expected behaviour is most definitely to stop at the `S` in String and then the `m` in `main`.  
No room for debate.

`Ctrl+Backspace` at least you are describing the standard behavior, and I agree it should be there.  I already don't use it though- I hold shift and ctrl and arrow for everything, then once selected hit backspace. This however is NOT working, and it should be.

**Comment by Ragnoroct** [comment]
I'm having the same problem. 
del: Deletes single character in front of cursor
backspace: Deletes whole word behind cursor
ctrl+backspace: Deletes single character behind cursor

Using a fresh version of clink on Windows 10 2004 I'm experiencing ctrl+backspace removing a single character and backspace removing a whole word.

I think this code is the culprit 
https://github.com/mridgers/clink/blob/823d841bc5c38a83c3cc0991c1f3ff8099d83fb5/clink/terminal/src/win_terminal_in.cpp#L258-L269

From reading that it sounds like the ctrl character is virtually always pressed and unset for specific keys. This makes sense why the `del (0x2E)` key works fine but `backspace (0x08)` doesn't since this if statement only evaluates true for characters greater than 0x1f.

**I would like to press backspace without the ctrl key and it **only** delete a single character.**

**Comment by chrisant996** [comment]
This is fixed in the [chrisant996/clink](https://github.com/chrisant996/clink) fork.
